op-supervisor/supervisor/backend/db/query.go

```diff
@@ -43,14 +45,40 @@ func (db *ChainsDB) LastCommonL1() (types.BlockSeal, error) {
 		if err != nil {
 			return types.BlockSeal{}, fmt.Errorf("failed to determine Last Common L1: %w", err)
 		}
-		// if the common block isn't yet set,
-		// or if the new common block is older than the current common block
-		// set the common block
 		if commonL1 == (types.BlockSeal{}) ||
 			last.Source.Number < commonL1.Number {
 			commonL1 = last.Source
 		}
 	}
+	// Step 2: offset to the parent block if completeOnly is true
+	if completeOnly {
+		// we only need one chain to offset to the parent block
+		ldb, ok := db.localDBs.Get(db.depSet.Chains()[0])
+		if !ok {
+			return types.BlockSeal{}, types.ErrUnknownChain
+		}
+		parent, err := ldb.SourceNumToSource(commonL1.Number - 1)
```

Comment:
Will this fail if the chain started at L1 genesis and that's all we've processed? Quite the corner case but still...

Reply:
For challenger it would be fine to just return the block currently being processed as long as it's guaranteed the parent is complete.  We just wait until it's strictly greater than the game L1 head instead of `>=` which we do for op-node anyway and is probably simpler than needing to look up the parent here.